code/game/machinery/camera/trackable.dm

```diff
@@ -0,0 +1,128 @@
+///How many ticks to try to find a target before giving up.
+#define CAMERA_TICK_LIMIT 10
+
+/datum/trackable
+	///Boolean on whether or not we are currently trying to track something.
+	var/tracking = FALSE
+	///Reference to the atom that owns us, used for tracking.
+	var/atom/tracking_holder
+
+	///If there is a mob currently being tracked, this will be the weakref to it.
+	var/datum/weakref/tracked_mob
+	///How many times we've failed to locate our target.
+	var/cameraticks = 0
+
+	///List of all names that can be tracked.
+	VAR_PRIVATE/list/names = list()
+	///List of all namecounts for mobs with the exact same name, just in-case.
+	VAR_PRIVATE/list/namecounts = list()
+	///List of all humans trackable by cameras.
+	VAR_PRIVATE/static/list/humans = list()
+	///List of all non-humans trackable by cameras, split so humans take priority.
+	VAR_PRIVATE/static/list/others = list()
+
+/datum/trackable/New(mob/source)
+	. = ..()
+	tracking_holder = source
+	RegisterSignal(tracking_holder, COMSIG_MOB_RESET_PERSPECTIVE, PROC_REF(cancel_target_tracking))
+
+/datum/trackable/Destroy(force, ...)
+	tracking_holder = null
+	tracked_mob = null
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/datum/trackable/process()
+	var/mob/living/tracked_target = tracked_mob?.resolve()
+	if(!tracked_target || !tracking)
+		set_tracking(FALSE)
+		return
+
+	if(tracked_target.can_track(tracking_holder))
+		cameraticks = initial(cameraticks)
+		SEND_SIGNAL(tracking_holder, COMSIG_TRACKABLE_TRACKING_TARGET, tracked_target)
+		return
+
+	if(cameraticks < CAMERA_TICK_LIMIT)
+		if(!cameraticks)
+			to_chat(tracking_holder, span_warning("Target is not near any active cameras. Attempting to reacquire..."))
+		cameraticks++
+		return
+
+	to_chat(tracking_holder, span_warning("Unable to reacquire, cancelling track..."))
+	cameraticks = initial(cameraticks)
+	set_tracking(FALSE)
+
+///Generates a list of trackable people by name, returning a list of Humans + Non-Humans that can be tracked.
+/datum/trackable/proc/trackable_mobs()
```

Comment:
Picky but can we call this `find_trackable_mobs` instead? `trackable_mobs` sounds like a var name.